common/src/main/java/io/netty/util/internal/InternalThreadLocalMap.java

```diff
@@ -163,17 +177,16 @@ public int size() {
     }
 
     public StringBuilder stringBuilder() {
-        final int stringBuilderCapacity = 1024;
-        if (stringBuilder == null) {
-            stringBuilder = new StringBuilder(stringBuilderCapacity);
-        } else {
-            if (stringBuilder.capacity() > stringBuilderCapacity) {
-                stringBuilder.setLength(stringBuilderCapacity);
-                stringBuilder.trimToSize();
-            }
-            stringBuilder.setLength(0);
+        StringBuilder sb = stringBuilder;
+        if (sb == null) {
+            return stringBuilder = new StringBuilder(STRING_BUILDER_INITIAL_SIZE);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract the assignment out of this expression. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)